pkg/template/controller/templatebrokeruuidmap_deletion_controller.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"time"
+
+	"github.com/openshift/origin/pkg/client"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// This controller watches for TemplateBrokerUUIDMap objects which are set to gracefully
+// delete, and deletes them when their time is up.
```

Comment:
Need to have a discussion about this specifically - graceful can be used to signal deletion in this fashion (synchronous GC will use it), but I'm not clear the intent yet.